### ISSUE TITLE
Error when a new revision aprovisioning

### ISSUE THREAD

**Issue by CristoC84** [microsoft/azure-container-apps#1217]
> Please provide us with the following information:
> ---------------------------------------------------------------

### This issue is a: (mark with an x)
- [x] bug report -> please search issues before submitting
- [ ] documentation issue or request
- [ ] regression (a behavior that used to work and stopped in a new release)

### Issue description
A clear and concise description of the observed issue.

This is a problem that occurs when new revisions are created automatically. The way in which we notice that there are problems with the container, is when for example we use postman and at the moment of triggering the query and after a few minutes the error **504 Gateway TimeOut** is generated or from a browser with the error **stream timeout**.
Internally the application seems to run smoothly but it is consumed externally. 
The clearest way to demonstrate this problem is to go to the **revisions and replicas** section within the same application container and in the running status section it maintains a fixed status of **Activating**, which I will point out later in images.

### Steps to reproduce
It is not clear to me how to reproduce the problem as it is quite random.

**Expected behavior**

The recommended approach would be to follow the Microsoft [documentation](https://learn.microsoft.com/en-us/azure/container-apps/revisions):

> In single revision mode, Container Apps ensures your app doesn't experience downtime when creating a new revision. The existing active revision isn't deactivated until the new revision is ready.

**Actual behavior** [What actually happened.]

When a new revision is provisioned, it is frozen and does not change to its ready state. This prevents queries from external networks.

**Screenshots**         

The red arrow is when the container was in trouble and the green arrow is when the container was restarted and started working normally.
![image](https://github.com/microsoft/azure-container-apps/assets/47341302/121c2fe0-453d-4936-bd09-bc2e7185e321)

And when I meant how do we validate that there is a problem with the container, it is by looking at the current status of the revision I mentioned in the issue description section.

![image](https://github.com/microsoft/azure-container-apps/assets/47341302/fa1d3bd6-d56c-4afe-b59e-e19e22fdb5fc)

It does not change its **Activating** status, until one restarts the container or creates a new revision.

Error when consulting the URL from a browser:

![image](https://github.com/microsoft/azure-container-apps/assets/47341302/7ed5d43d-b99c-412b-aa29-bc4864100069)

### Additional context

- The application only works on port **3000**, no other port.
- The Ingress configuration is **enabled**, accepts traffic from anywhere, ingress type **HTTP**, client certificate mode **Ignore**, transport **Auto**, Insecure connections **disabled** and target port **3000**.
- The Container Apps Environment internal **false**.
- From the Dockerfile, **port 3000 and 22 are exposed**, but the latter is not used.
- I do not use probes.

**Comment by fritzfs** [comment]
@CristoC84 any luck solving this? We encountered [this issue](https://github.com/microsoft/azure-container-apps/issues/1219) now :-(

**Comment by CristoC84** [comment]
hi @fritzfs 
After raising a ticket with Microsoft they gave me signs as to what was going on.

The application, as mentioned above, worked only through port 3000 so the health probes scanned using their default values.

Then when there was some change in the application and a new revision was generated, this never materialized because the application did not respond under any http/https port causing a freeze in changing the status to ready.
The startup probe is modified to TCP and it has not generated again this freezing in the creation of new revisions.
Note that I only changed the startup probe, the others, liveness and readiness probes were left by default.
Another important thing to point out and I think it is relevant, is that the ingress type I have it as HTTP when it should be as TCP, but I left it that way because it allows me to access the application under HTTPS and avoid having to add at the end of the Application URL the port ":3000".
I hope I have helped someone with this problem and it would be good to add this in the FAQ or Troubleshotting about the importance of defining the Ingress Type and about the health probes.
